app/src/main/java/com/davidtomas/taskyapp/features/auth/domain/useCase/ValidatePasswordUseCase.kt

```diff
@@ -0,0 +1,19 @@
+package com.davidtomas.taskyapp.features.auth.domain.useCase
+
+import com.davidtomas.taskyapp.features.auth.domain.model.LoginInputValidationError
+import com.davidtomas.taskyapp.features.auth.domain.model.MissingMandatoryField
+import com.davidtomas.taskyapp.features.auth.domain.model.PasswordFormatError
+
+class ValidatePasswordUseCase {
+    operator fun invoke(password: String): LoginInputValidationError? =
+        when {
+            password.isEmpty() -> MissingMandatoryField
+            !password.matches(PASSWORD_REGEX.toRegex()) -> PasswordFormatError
+            else -> null
+        }
+    companion object {
+        private const val MIN_PASSWORD_LENGTH = 9
+        const val PASSWORD_REGEX =
+            "^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@\$ %^&*-]).{$MIN_PASSWORD_LENGTH,}\$"
+    }
+}
```

Comment:
For simple patterns like a valid password, I'm always a friend of doing this manually since it lets you quickly know what makes a valid password. Something like this: 
```
val hasLowerCase = password.any { it.isLowerCase() }
val hasUpperCase = password.any { it.isUpperCase() }
val hasDigit = password.any { it.hasDigit() }
val hasValidLength = password.length in min..max

return hasLowerCase && hasUpperCase && hasDigit && hasValidLength
```
That way, probably even my grandma could tell you what the requirements for a valid password are :D 
Regex makes sense if you can't easily do this manually, but has the cost of making code very unreadable.